momentum_scanner_intraday.py

```diff
@@ -162,43 +157,46 @@ def get_high_returns(
 ):
     vols = query_volumes(dex)
     lag_col = f"{lag_return}H Return"
-    df = pd.DataFrame()
-    if len(vols) != 0:
-        vols1 = filter_pairs(vols, volume=daily_volume)
-        if len(vols1) > 0:
-            df = find_rets_24h(vols1)
-            df = add_volume_marketcap(df)
-            df = df[
-                (df["30_day_mean_volume"] >= vol_30)
-                & (df["30_day_mean_marketcap"] >= market_cap)
-            ]
-            if len(df) == 0:
-                return df
-            df = df[df["24H Return"] >= 0]
-            df = add_7drets(df)
-            df = add_intraday_rets(df, lag_return)
-            df[lag_col] = df[lag_col].apply(lambda x: round(x * 100, 2))
-            df = df[df[lag_col] >= 0]
-            df = df.sort_values(by=lag_col, ascending=False)
+    vols = filter_pairs(vols, volume=daily_volume)
+    df = find_rets_24h(vols)
+    df = add_volume_marketcap(df)
+    df = df[
+        (df["30_day_mean_volume"] >= vol_30)
+        & (df["30_day_mean_marketcap"] >= market_cap)
+    ]
+    if df.empty:
+        return df
+    df = df[df["24H Return"] >= 0]
+    df = add_7drets(df)
+    df = add_intraday_rets(df, lag_return)
+    df[lag_col] = df[lag_col].apply(lambda x: round(x * 100, 2))
+    df = df[df[lag_col] >= 0]
+    df = df.sort_values(by=lag_col, ascending=False)
+
     return df
 
 
+def required_pairs(dex):
+    return pd.DataFrame(
+        [(f"{a}<>{b}", 1e7) for a, b in cache_db.get_pairs(dex)],
+        columns=("pair", "volume"),
+    ).set_index("pair")
+
+
 def find_best_return(dex, stoploss, profittaking, lag):
     lag_col = f"{lag}H Return"
     vols = query_volumes(dex)
-    if len(vols) != 0:
-        vols1 = filter_pairs(vols, volume=150000)
-        if len(vols1) == 0:
-            print("No pair found with enough volume")
-            return
-        else:
-            vols1 = vols1  # .iloc[1:]
-    else:
-        print("Endpoint issues, query did not get any returned values")
-        return
+    if vols.empty:
+        raise Exception("Query did not get any returned values")
+
+    vols = filter_pairs(vols, volume=150000)
+    if vols.empty:
+        raise Exception("No pair found with enough volume")
 
-    vols1 = add_tokens(dex, vols1)
-    df = find_rets_24h(vols1)
+    extras = required_pairs(dex)
```

Comment:
Note that even though we use this here, we don't use it in `get_high_returns`. @fieldtheory123

Reply:
Noted